iree/test/e2e/tosa_ops/if.mlir

```diff
@@ -0,0 +1,35 @@
+func @if_true_test() attributes { iree.module.export } {
+  %0 = iree.unfoldable_constant dense<true> : tensor<i1>
+  %1 = iree.unfoldable_constant dense<5> : tensor<i32>
+  %2 = "tosa.cond_if"(%0, %1) ( {
+  ^bb0(%arg2: tensor<i32>):  // no predecessors
+    %c1 = constant dense<1> : tensor<i32>
+    %2 = "tosa.add"(%arg2, %c1) : (tensor<i32>, tensor<i32>) -> tensor<i32>
+    "tosa.yield"(%2) : (tensor<i32>) -> ()
+  },  {
+  ^bb0(%arg2: tensor<i32>):  // no predecessors
+    %c1 = constant dense<2> : tensor<i32>
+    %3 = "tosa.add"(%arg2, %c1) : (tensor<i32>, tensor<i32>) -> tensor<i32>
+    "tosa.yield"(%3) : (tensor<i32>) -> ()
```

Comment:
Rather than add a different number in the unexpected case, do we have a way to assert / check fail?

Or, how about adding a very large number to make it especially clear that this branch isn't expected to be taken? The way you have this now does contrast nicely with the false test though. Up to you.

Reply:
Guarantee the correct branch executes via some included math, along with a failure case on the not case.